src/test/java/com/google/cloud/spanner/jdbc/it/ITProtoColumnsTest.java

```diff
@@ -0,0 +1,218 @@
+                        + "spanner.examples.music.Genre,"
+                        + ")",
+                    "CREATE TABLE Types ("
+                        + "  RowID INT64 NOT NULL,"
+                        + "  ProtoMessage    spanner.examples.music.SingerInfo,"
+                        + "  ProtoEnum   spanner.examples.music.Genre,"
+                        + "  ProtoMessageArray   ARRAY<spanner.examples.music.SingerInfo>,"
+                        + "  ProtoEnumArray  ARRAY<spanner.examples.music.Genre>,"
+                        + "  ) PRIMARY KEY (RowID)"))
+            .get(OPERATION_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+
+    in.close();
+    url = "jdbc:cloudspanner://" + host + "/" + database.getId();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    if (database != null) {
+      database.drop();
+    }
+    ConnectionOptions.closeSpanner();
+  }
+
+  @Test
+  public void testNonNullElements() throws Exception {
+    assumeFalse(
+        "Proto columns is not supported in the emulator", EmulatorSpannerHelper.isUsingEmulator());
+    SingerInfo singerInfo =
+        SingerInfo.newBuilder()
+            .setSingerId(1)
+            .setNationality("Country1")
+            .setGenre(Genre.ROCK)
+            .build();
+    Genre singerGenre = Genre.ROCK;
+    SingerInfo[] singerInfoArray = new SingerInfo[] {singerInfo, null};
+    Genre[] singerGenreArray = new Genre[] {Genre.ROCK, null};
+    try (Connection connection = DriverManager.getConnection(url);
+        PreparedStatement ps =
+            connection.prepareStatement(
+                "INSERT INTO Types"
+                    + " (RowID, ProtoMessage, ProtoEnum, ProtoMessageArray, ProtoEnumArray) VALUES (?, ?, ?, ?, ?)")) {
+      ps.setInt(1, 1);
+      ps.setObject(2, singerInfo, ProtoMessageType.INSTANCE);
+      ps.setObject(3, singerGenre, ProtoEnumType.INSTANCE);
+      ps.setArray(4, connection.createArrayOf("PROTO", singerInfoArray));
+      ps.setArray(5, connection.createArrayOf("ENUM", singerGenreArray));
+
+      final int updateCount = ps.executeUpdate();
+      assertEquals(1, updateCount);
+    }
+
+    try (Connection connection = DriverManager.getConnection(url);
+        Statement statement = connection.createStatement();
+        ResultSet resultSet = statement.executeQuery("SELECT * FROM Types WHERE RowID = 1")) {
+
+      resultSet.next();
+      assertEquals(1, resultSet.getInt("RowID"));
+      assertEquals(singerInfo, resultSet.getObject("ProtoMessage", SingerInfo.class));
+      assertEquals(singerGenre, resultSet.getObject("ProtoEnum", Genre.class));
+      assertArrayEquals(
+          singerInfoArray, resultSet.getObject("ProtoMessageArray", SingerInfo[].class));
+      assertArrayEquals(singerGenreArray, resultSet.getObject("ProtoEnumArray", Genre[].class));
+
+      assertThat(resultSet.next(), is(false));
+    }
+  }
+
+  @Test
+  public void testNullElements() throws Exception {
```

Comment:
Could we test this with untyped null values as well? So just calling `ps.setObject(2, null)` etc.

Reply:
Done

Reply:
Thanks